[DA-2971] updating the retention cloud function to check if 'retention' in file

ayazamlani opened the pull request:
## Resolves DA-2971 
[ticket](https://precisionmedicineinitiative.atlassian.net/browse/DA-2971?atlOrigin=eyJpIjoiYTUxNmJiZmNmZTdkNDIwYWI5NWNjMGZkODhhNTZhYjUiLCJwIjoiaiJ9)

## Description of changes/additions
Updating the retention cloud function to check if 'retention' in filename


## Tests

- [ ] Unit Tests
- 1 Failed -> cryptography, installed 3.3.2, affected <39.0.0

ayazamlani committed:
updating the retention cloud function to check if 'retention' in filename

ayazamlani committed:
changing trailing error found in codeacy, also fixing param name in docstring to avoid warnings that occur in pycharm

dtharpeuno commented on rdr_service/gcloud_functions/retention_eligible_import_function/main.py:
```diff
@@ -78,12 +78,17 @@ def retention_eligible_import_function(_event, _context):
-        func = RetentionEligibleImportFunction(gcp_env, _event, _context)
-        func.run()
+        # checking to see if the file being dropped in the bucket is the retention file before executing the function
+        file_name = _event['name']
+        if 'retention' in file_name:
+            func = RetentionEligibleImportFunction(gcp_env, _event, _context)
+            func.run()
+        else:
+            _logger.info(f"skipping file: {file_name}, as it is not the retention file.")
```
We generally don't want to short-circuit the class here in the Cloud function since this function can be used for a variety of file types. The `created` method is where this logic should live since we can build on that later.

ayazamlani committed:
## Resolves DA-2971
https://precisionmedicineinitiative.atlassian.net/browse/DA-2971?atlOrigin=eyJpIjoiYTUxNmJiZmNmZTdkNDIwYWI5NWNjMGZkODhhNTZhYjUiLCJwIjoiaiJ9

## Description of changes/additions
Moving check if 'retention' in filename to created function

## Tests

- [ ] Unit Tests

dtharpeuno commented on rdr_service/gcloud_functions/retention_eligible_import_function/main.py:
```diff
@@ -42,10 +42,14 @@ def created(self):
+            _task = GCPCloudTask()
+            _task.execute('/resource/task/ImportRetentionEligibleFileApi', payload=data, queue=task_queue)
+        else:
+            _logger.info(f"skipping file: {self.event.name}, as it is not the retention file.")
```
:nit capitalize, please
